compaction/task_manager_module.cc

```diff
@@ -299,6 +309,7 @@ future<> shard_major_keyspace_compaction_task_impl::run() {
 future<> table_major_keyspace_compaction_task_impl::run() {
     co_await wait_for_your_turn(_cv, _current_task, _status.id);
     tasks::task_info info{_status.id, _status.shard};
+    replica::table::do_flush flush(_flush_mode == flush_mode::compacted_tables);
```

Comment:
What about `all_tables`?

Reply:
`all_tables` is handled bu global/keyspace tasks.
In case it was set and the decided to skip it since the previous flush_all_tables happened not long enough ago, they convert it downstream to `compacted_tables`, see above:
```
flush_mode fm = _flush_mode == flush_mode::skip || flushed_all_tables ? flush_mode::skip : flush_mode::compacted_tables;
```

Reply:
We can also change it here to `_flush_mode != flush_mode::skip` for simplicity.

Reply:
And then the above would look like `flush_mode fm = flushed_all_tables ? flush_mode::skip : _flush_mode;

Reply:
done the above in next version.